File: ProspectManagerWebApi/Models/Prospect.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace ProspectManagerWebApi.Models
{
    [Index(nameof(Nom), IsUnique = true)]
    public class Prospect : ITableHistorique
    {
        public int Id { get; set; }

        [Required]
        public string? Nom { get; set; } = "";
        public string? Departement { get; set; }
        public string? Adresse { get; set; }
        public string? Telephone { get; set; }
        public string? Mail { get; set; }
        public string? SecteurActivite { get; set; }
        public DateTimeOffset DateCreation { get; set; }
        public Statut Statut { get; set; } = new Statut();
        public bool Actif {  get; set; }
        public TypeOrganisme TypeOrganisme { get; set; } = new TypeOrganisme();
        public ICollection<Contact> Contacts { get; set; } = new List<Contact>();
        public Utilisateur UtilisateurCreation { get; set; } = new Utilisateur();
        public ICollection<Modification> Modifications { get; set; } = new List<Modification>();
        public ICollection<ProduitProspect> ProduitProspects { get; set; } = new List<ProduitProspect>();
        public ICollection<Evenement> Evenements { get; set; } = new List<Evenement>();
    }
}

File: ProspectManagerClient/src/app/modules/gestion-prospect/models/prospect.ts
import { Contact } from "./contact";
import { Evenement } from "./evenement";
import { Modification } from "./modification";
import { ProduitProspect } from "./produitprospect";
import { SecteurGeographique } from "./secteur-geographique";
import { Statut } from "./statut";
import { TypeOrganisme } from "./type-organisme";
import { Utilisateur } from "./utilisateur";

export class Prospect {
    id: number;
    nom: string = "";
    secteurGeographique?: SecteurGeographique;
    adresse: string = "";
    telephone: string = "";
    mail: string = "";
    secteurActivite: string = "";
    dateCreation: string = "";
    statut?: Statut;
    typeOrganisme?: TypeOrganisme;
    contacts: Contact[] = [];
    utilisateurCreation?: Utilisateur;
    modifications: Modification[];
    produitProspects: ProduitProspect[] = [];
    evenements: Evenement[] = [];
}